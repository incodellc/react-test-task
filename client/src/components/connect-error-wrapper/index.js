import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;