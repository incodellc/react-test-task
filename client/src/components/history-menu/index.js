import HistoryMenu from './HistoryMenu';
export default HistoryMenu;