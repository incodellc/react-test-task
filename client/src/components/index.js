import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };