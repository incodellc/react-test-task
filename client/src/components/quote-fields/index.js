import QuoteFields from './QuoteFields';
export default QuoteFields;