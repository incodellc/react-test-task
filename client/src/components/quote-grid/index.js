import QuotesGrid from './QuotesGrid';
export default QuotesGrid;