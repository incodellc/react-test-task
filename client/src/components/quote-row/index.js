import QuoteRow from './QuoteRow';
export default QuoteRow;