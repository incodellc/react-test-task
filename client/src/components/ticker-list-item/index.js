import TickerListItem from './TickerListItem';
export default TickerListItem;