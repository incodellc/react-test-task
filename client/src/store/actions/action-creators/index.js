import actionTypes from '../action-types';
import { tickerService } from '../../../services';
import { tickers } from '../../../utils';

export const streamServerConnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_CONNECT
    };
};

export const streamServerDisconnect = () => {
    return {
        type: actionTypes.STREAM_SERVER_DISCONNECT
    };
};

export const updateQuotes = (data) => {
    return {
        type: actionTypes.UPDATE_QUOTES,
        data
    };
};

export const connectToStreamServer = () => {
    return dispatch => {
        dispatch({
            type: actionTypes.CONNECTION_INIT
        });
        tickerService(
            tickers, 
            () => dispatch(streamServerConnect()), 
            (data) => dispatch(updateQuotes(data)),
            () => dispatch(streamServerDisconnect())
        );
    };
};