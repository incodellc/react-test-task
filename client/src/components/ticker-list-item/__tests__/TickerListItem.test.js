import React from 'react';
import { tickers } from '../../../utils';
import { renderWithRedux } from '../../../utils/test-utils';
import { TickerListItem } from '../../../components';
import { toggleTicker } from '../../../store/actions/action-creators';
import { cleanup, fireEvent } from '@testing-library/react';

describe('Check TickerListItem component', () => {
    afterEach(cleanup);

    it('component renders correctly', () => {
        const { asFragment } = renderWithRedux(<TickerListItem ticker={tickers[0]} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('component should have correct text content', () => {
        const { getByTestId } = renderWithRedux(<TickerListItem ticker={tickers[0]} />);
        const element = getByTestId('ticker-list-item');
        expect(element).toHaveTextContent('AAPL');
        expect(element).toHaveTextContent('Apple Inc.');
        expect(element).toHaveTextContent('213.41');
        expect(element).toHaveTextContent('-38.72');
    });

    it('component should display negative changes correctly', () => {
        const { getByTestId } = renderWithRedux(<TickerListItem ticker={tickers[0]} />);
        expect(getByTestId('ticker-list-item-negative')).toBeInTheDocument();
        expect(getByTestId('ticker-list-item-icon-down')).toHaveClass('arrow down');
    });

    it('component should display positive changes correctly', () => {
        const { getByTestId } = renderWithRedux(<TickerListItem ticker={tickers[1]} />);
        expect(getByTestId('ticker-list-item-positive')).toBeInTheDocument();
        expect(getByTestId('ticker-list-item-icon-up')).toHaveClass('arrow up');
    });
    
    it('component should receive updated state and react correctly after click event fired', () => {
        const { getByTestId, store } = renderWithRedux(
            <TickerListItem 
                ticker={tickers[0]}
                onTickerClick={(e, { id }) => store.dispatch(toggleTicker(id))}
            />
        );
        expect(getByTestId('ticker-list-item')).not.toHaveClass('active');
        fireEvent.click(getByTestId('ticker-list-item'));
        expect(getByTestId('ticker-list-item')).toHaveClass('active');
    });
});
