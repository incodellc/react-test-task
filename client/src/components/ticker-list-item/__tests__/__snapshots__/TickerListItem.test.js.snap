// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check TickerListItem component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="item ui card tickerListItem"
    data-testid="ticker-list-item"
    id="AAPL"
    role="listitem"
  >
    <div
      class="content"
    >
      <div
        class="glow right floated animating transition"
        data-testid="ticker-list-item-negative"
        style="animation-duration: 2000ms;"
      >
        <div
          class="ui mini statistic"
        >
          <div
            class="value"
          >
            213.41
          </div>
          <div
            class="label"
          >
            <span
              class="pricedown"
            >
              -38.72
            </span>
          </div>
        </div>
        <i
          aria-hidden="true"
          class="red arrow down big icon"
          data-testid="ticker-list-item-icon-down"
        />
      </div>
      <div
        class="header"
      >
        AAPL
      </div>
      <div
        class="meta"
      >
        Apple Inc.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
